sql/sql_delete.cc

```diff
@@ -869,6 +869,44 @@ int Query_result_delete::prepare(List<Item> &values, SELECT_LEX_UNIT *u)
 }
 
 
+/**
+  Test that the subject table (of DELETE) has a cascade foreign key
+  parent present in the query.
+
+  @param  thd        thread handle
+  @param  table      table to be checked (must be updatable base table)
+  @param  table_list List of tables to check against
+
+  @retval NULL     No cascade parents found.
+*/
+static bool has_cascade_dependency(THD *thd, const TABLE_LIST &table,
+                                   TABLE_LIST *table_list)
+{
+  DBUG_ASSERT(&table == const_cast<TABLE_LIST &>(table).updatable_base_table());
+
+  List <st_handler_tablename> fk_table_list;
+  List_iterator<st_handler_tablename> fk_table_list_it(fk_table_list);
+
+  table.table->file->get_cascade_foreign_key_table_list(thd, &fk_table_list);
+
+  st_handler_tablename *tbl_name;
+  while ((tbl_name= fk_table_list_it++))
+  {
+    for (TABLE_LIST* curr= table_list; curr; curr= curr->next_local)
+    {
+      const bool same_table_name= lower_case_table_names ?
+        my_strcasecmp(files_charset_info, curr->table_name, tbl_name->tablename) :
+        strcmp(curr->table_name, tbl_name->tablename);
+      if (same_table_name && strcmp(curr->db, tbl_name->db) == 0)
```

Comment:
As far as I can tell by existing use, lower_case_table_names govern DB names too.
Also I see some inconsistency in existing code whether files_charset_info or system_charset_info should be used, please double-check which is which.